test/swar/BasicOperations.cpp

```diff
@@ -6,15 +6,146 @@
+static_assert(4 == popcount<4>(0xF00ull));
+static_assert(8 == popcount<4>(0xFF0ull));
+static_assert(9 == popcount<4>(0xEEEull));
+static_assert(0x210 == popcount<0>(0x320));
+static_assert(0x4321 == popcount<1>(0xF754));
+static_assert(0x50004 == popcount<3>(0x3E001122));
+
+static_assert(1 == msb_index<u64>(1ull<<1));
+static_assert(3 == msb_index<u64>(1ull<<3));
+static_assert(5 == msb_index<u64>(1ull<<5));
+static_assert(8 == msb_index<u64>(1ull<<8));
+static_assert(17 == msb_index<u64>(1ull<<17));
+static_assert(30 == msb_index<u64>(1ull<<30));
+static_assert(31 == msb_index<u64>(1ull<<31));
+
+static_assert(0xAA == makeBitmask<2, u8>(0x2));
+static_assert(0x0808'0808ull == makeBitmask<8, u32>(0x8));
+static_assert(0x0808'0808'0808'0808ull == makeBitmask<8, u64>(0x08ull));
+static_assert(0x0101'0101'0101'0101ull == makeBitmask<8, u64>(0x01ull));
+static_assert(0x0E0E'0E0E'0E0E'0E0Eull == makeBitmask<8, u64>(0x0Eull));
+static_assert(0x0303'0303'0303'0303ull == makeBitmask<8, u64>(0x03ull));
+
+static_assert(0x00 == clearLSB<u8>(0x80));
+static_assert(0x80 == clearLSB<u8>(0xC0));
+static_assert(0xC0 == clearLSB<u8>(0xE0));
+static_assert(0xE0 == clearLSB<u8>(0xF0));
+static_assert(0xF0 == clearLSB<u8>(0xF8));
+static_assert(0xF8 == clearLSB<u8>(0xFC));
+static_assert(0xFC == clearLSB<u8>(0xFE));
+static_assert(0xFE == clearLSB<u8>(0xFF));
+static_assert(0x00 == clearLSB<u8>(0x00));
+
+static_assert(0x00 == isolateLSB<u8>(0x00));
+static_assert(0x10 == isolateLSB<u8>(0xF0));
+static_assert(0x20 == isolateLSB<u8>(0xE0));
+static_assert(0x40 == isolateLSB<u8>(0xC0));
+static_assert(0x80 == isolateLSB<u8>(0x80));
+
+static_assert(0x01 == maskLowBits<1, u8>());
+static_assert(0x03 == maskLowBits<2, u8>());
+static_assert(0x07 == maskLowBits<3, u8>());
+static_assert(0x0F == maskLowBits<4, u8>());
+
+static_assert(0xB == isolate<4>(0x1337'BDBC'2448'ACABull));
+static_assert(0xAB == isolate<8>(0x1337'BDBC'2448'ACABull));
+static_assert(0xCAB == isolate<12>(0x1337'BDBC'2448'ACABull));
+static_assert(0xACAB == isolate<16>(0x1337'BDBC'2448'ACABull));
+static_assert(0x3 == isolate<3>(0x1337'BDBC'2448'ACABull));
+
+static_assert(0x00 == clearLSBits<2, u8>(0x80));
+static_assert(0x00 == clearLSBits<2, u8>(0xC0));
+static_assert(0x80 == clearLSBits<2, u8>(0xE0));
+static_assert(0xC0 == clearLSBits<2, u8>(0xF0));
+static_assert(0xE0 == clearLSBits<2, u8>(0xF8));
+static_assert(0xF8 == clearLSBits<2, u8>(0xFB));
+static_assert(0xF0 == clearLSBits<2, u8>(0xFC));
+static_assert(0xFC == clearLSBits<2, u8>(0xFF));
+
+static_assert(0x80 == isolateLSBits<2, u8>(0x80));
+static_assert(0xC0 == isolateLSBits<2, u8>(0xC0));
+static_assert(0x60 == isolateLSBits<2, u8>(0xE0));
+static_assert(0x30 == isolateLSBits<2, u8>(0xF0));
+static_assert(0x18 == isolateLSBits<2, u8>(0xF8));
+static_assert(0x03 == isolateLSBits<2, u8>(0xFB));
+static_assert(0x0C == isolateLSBits<2, u8>(0xFC));
+static_assert(0x03 == isolateLSBits<2, u8>(0xFF));
```

Comment:
I think the suffix `Bits` does not add over `Bs`

Reply:
Sorry, but I don't like using the element size of 2 for these tests, the convenient way to do this, I htik is to use nibbles, 4 bits.

Reply:
I wanted to use minimal size testing. Truly I should add 2,3,4,etc bits to various sizes to be truly paranoid in testing.